crates/sui-core/src/validator_tx_finalizer.rs

```diff
@@ -0,0 +1,426 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::AuthorityState;
+use crate::authority_aggregator::AuthorityAggregator;
+use crate::authority_client::AuthorityAPI;
+use std::sync::{Arc, Weak};
+use std::time::Duration;
+use sui_types::transaction::VerifiedSignedTransaction;
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, error};
+
+/// Only wake up the transaction finalization task for a given transaction
+/// after 15 mins of seeing it. This gives plenty of time for the transaction
+/// to become final in the normal way.
+const TX_FINALIZATION_DELAY: Duration = Duration::from_secs(15 * 60);
+/// If a transaction can not be finalized within 10 mins of being woken up, give up.
+const FINALIZATION_TIMEOUT: Duration = Duration::from_secs(10 * 60);
```

Comment:
Does having a long timeout help? Wouldn't AuthorityAggregator normally finish pretty quickly one way or the other? (it doesn't do any retries, right?)

With both this and the previous timeout, my concern is how much memory we might commit to pending tasks during sustained high load.